### PR TITLE
[Feature] support create table ... as select ...

### PR DESCRIPTION
**Pull request by ck89119**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue https://github.com/matrixorigin/matrixone/issues/10362

## What this PR does / why we need it:
[Feature] support create table ... as select ...

### CODE REVIEW

**Review by iamlinjunhong — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/compile/ddl.go [hunk 1]
```diff
@@ -1112,12 +1112,28 @@ func (s *Scope) CreateTable(c *Compile) error {
...
 		c.proc.TxnOperator,
 		nil)
+	if err != nil {
+		return err
+	}
+
+	if stmt, ok := c.stmt.(*tree.CreateTable); ok && stmt.IsAsSelect {
+		getLogger().Info("createTable", zap.String("CreateAsSelectSql", qry.CreateAsSelectSql))
+		if err = c.runSql(qry.CreateAsSelectSql); err != nil {
+			getLogger().Info("createTable",
```
Comment (zhangxu19830126) [inline pkg/sql/compile/ddl.go:1128]: log error, not log info

File: pkg/sql/compile/ddl.go [hunk 2]
```diff
@@ -1112,12 +1112,28 @@ func (s *Scope) CreateTable(c *Compile) error {
...
 		c.ctx,
 		dbSource,
 		qry.GetTableDef(),
 		c.proc.TxnOperator,
 		nil)
+	if err != nil {
+		return err
+	}
+
+	if stmt, ok := c.stmt.(*tree.CreateTable); ok && stmt.IsAsSelect {
```
Comment (zhangxu19830126) [inline pkg/sql/compile/ddl.go:1125]: move this logic before create auto incr

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

File: pkg/sql/compile/ddl.go (hunk 2 above)
Comment (ck89119) [reply pkg/sql/compile/ddl.go:1125]: 如果新表中有自增列，这段插入数据的逻辑如果放在新建自增列前面会报错

``` sql
mysql> CREATE table t2 (a bigint unsigned not null auto_increment primary key, c float) as select a, b from t1;
ERROR 1146 (HY000): no such table test.t2
```

File: pkg/sql/compile/ddl.go (hunk 1 above)
Comment (ck89119) [reply pkg/sql/compile/ddl.go:1128]: 这边没有报错，我就是想把这段插入数据的sql打印出来

**Review by zhangxu19830126 — APPROVED** [review]

**Pull request merged**